source/fundamentals/auth.txt

```diff
@@ -10,22 +10,254 @@ Authentication Mechanisms
+
+   * - ``MONGODB-CR``
+     - MongoDB 3.6 and earlier
+
+To specify the default authentication mechanism, omit the
+``AuthMechanism`` option:
+
+.. code-block:: go
+
+   credential := options.Credential{
+      AuthSource: "<authenticationDb>",
+      Username: "<username>",
+      Password: "<password>",
+   }
+   clientOpts := options.Client().ApplyURI("mongodb://<hostname>:<port>").
+      SetAuth(credential)
+
+   client, err := mongo.Connect(context.TODO(), clientOpts)
+
+For more information on the challenge-response (CR) and salted
+challenge-response authentication mechanisms (SCRAM) that MongoDB supports,
+see the :manual:`SCRAM </core/security-scram/>` section of the server manual.
+
+``SCRAM-SHA-256``
+~~~~~~~~~~~~~~~~~
+
+.. important::
+
+   ``SCRAM-SHA-256`` is the default authentication method for MongoDB starting
+   in MongoDB 4.0.
+
+``SCRAM-SHA-256`` is a salted challenge-response authentication mechanism
+(SCRAM) that uses your username and password, encrypted with the ``SHA-256``
+algorithm, to authenticate your user.
+
+To specify the ``SCRAM-SHA-256`` authentication mechanism, assign the
+``AuthMechanism`` option the value ``SCRAM-SHA-256``:  
+
+.. code-block:: go
+   :emphasize-lines: 2
+
+   credential := options.Credential{
+      AuthMechanism: "SCRAM-SHA-256",
+      AuthSource: "<authenticationDb>",
+      Username: "<username>",
+      Password: "<password>",
+   }
+   clientOpts := options.Client().ApplyURI("mongodb://<hostname>:<port>").
+      SetAuth(credential)
+   
+   client, err := mongo.Connect(context.TODO(), clientOpts)
+
+.. _scram-sha-1-auth-mechanism:
+
+``SCRAM-SHA-1``
+~~~~~~~~~~~~~~~
+
+.. important::
+
+   ``SCRAM-SHA-1`` is the default authentication method for MongoDB versions
+   3.0, 3.2, 3.4, and 3.6.
+
+``SCRAM-SHA-1`` is a salted challenge-response mechanism (SCRAM) that uses your
+username and password, encrypted with the ``SHA-1`` algorithm, to authenticate
+your user.
+
+To specify the ``SCRAM-SHA-1`` authentication mechanism, assign the
+``AuthMechanism`` option the value ``SCRAM-SHA-1``:  
+
+.. code-block:: go
+   :emphasize-lines: 2
+
+   credential := options.Credential{
+      AuthMechanism: "SCRAM-SHA-1",
+      AuthSource: "<authenticationDb>",
+      Username:      "<username>",
+      Password:      "<password>",
+   }
+   clientOpts := options.Client().ApplyURI("mongodb://<hostname>:<port>").
+      SetAuth(credential)
+
+   client, err := mongo.Connect(context.TODO(), clientOpts)
+
+``MONGODB-CR``
+~~~~~~~~~~~~~~
+
+``MONGODB-CR`` is a challenge-response authentication mechanism that uses your
+username and password to authenticate your user. This authentication
+mechanism was deprecated starting in MongoDB 3.6 and is no longer
+supported as of MongoDB 4.0.
+
+You cannot specify this method explicitly; refer to the fallback provided
+by the :ref:`default authentication mechanism <default-auth-mechanism>` to
+connect using ``MONGODB-CR``.
```

Comment:
I believe you can specify `MONGODB-CR` in the Go driver by passing `"MONGODB-CR"` as the `AuthMechanism`.